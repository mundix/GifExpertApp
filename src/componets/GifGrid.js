import React, { useState, useEffect } from 'react'
import { getGifs } from '../Helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( {category}) => {

    const [images, setImages] = useState([]);
    
    useEffect( () => {
        getGifs(category)
        .then( setImages); //lo mismo de abajo
        // .then( imgs => setImages(imgs));
    }, []);

    return (
        <>
            <h3>{ category }</h3>
            <div className='card-grid'>
                    {
                        images.map(img => (
                            <GifGridItem 
                            {...img}
                            key={img.id}/>
                        ))
                    }
            </div>
        </>
    )
}
