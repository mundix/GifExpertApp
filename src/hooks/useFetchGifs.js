// Se puede hacer con un rfac y borrar para dejarlo asi.
import { useState, useEffect } from "react";
import {getGifs} from '../Helpers/getGifs';

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading:true
    });
    //los efectos no pueden ser async por que esperan algo sincrono
    useEffect(() => {

        //Aqui se hace la peticion y traer las imagenes
        getGifs(category)
        .then( imgs => {

            setTimeout(() => {
                setState({
                    data: imgs,
                    loading: false
                });
            }, 3000);

            
        });

    }, [category]);

    

    return state; //retorna le objeto de arriba {data:[], loading:true}
}
