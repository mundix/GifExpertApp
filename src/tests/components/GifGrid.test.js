import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../componets/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
//Si hago esto en cada test debo usar esta funcion falsead
jest.mock('../../hooks/useFetchGifs'); //esto ayuda a usar el useFetch y controlar la informacion quee sto va a responder 


describe('Pruebas en el archivo <GifGrid/>', () => {

    const category = 'Batman';

    test('Mostrando contenido', () => {
      
          useFetchGifs.mockReturnValue({
            data: [],
             loading: true
          });

        const wrapper =  shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        
    })
    
    //probar mi useFetch que debe mostar item cuanod se cargan imagenes 
    test('Debe de mostar items cuando se carga imagenes useFetch', async () => {
        //probar de mi custom hook regreso la informacion cuando ya la cargo. 
        // useFetchGifs
        const gifs = [
            {
              id: "ABC",
              url: "https://localhost/img.jpg",
              title: "Title",
            },
            {
              id: "123",
              url: "https://localhost/img.jpg",
              title: "Title",
            },
          ];
      
          useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
          });

        const wrapper =  shallow(<GifGrid category={category}/>);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);


    })
    

});
