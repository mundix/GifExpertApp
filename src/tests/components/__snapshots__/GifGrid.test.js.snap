// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo <GifGrid/> Debe de mostar items cuando se carga imagenes useFetch 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Batman
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Title"
      url="https://localhost/img.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Title"
      url="https://localhost/img.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el archivo <GifGrid/> Mostrando contenido 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Batman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
